console/src/components/Services/Data/Add/CheckConstraints.js

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useState } from 'react';
+import AceEditor from 'react-ace';
+
+import ExpandableEditor from '../../../Common/Layout/ExpandableEditor/Editor';
+import styles from '../TableModify/ModifyTable.scss';
+import { removeCheckConstraint, setCheckConstraints } from './AddActions';
+import ToolTip from '../../../Common/Tooltip/Tooltip';
+import KnowMoreLink from '../../../Common/KnowMoreLink/KnowMoreLink';
+
+const CheckConstraints = ({ dispatch, constraints }) => {
+  const [addConstraintsState, setAddConstraintsState] = useState([]);
+
+  useEffect(() => {
+    const newConstraintsState = [...constraints, { name: '', check: '' }];
+    setAddConstraintsState(newConstraintsState);
+  }, [constraints]);
+
+  return addConstraintsState.map(({ name, check }, i) => {
+    const onChangeName = e => {
+      e.persist();
+      setAddConstraintsState(prev =>
+        prev.map((c, idx) => {
+          if (i === idx) return { ...c, name: e.target.value };
+          return c;
+        })
+      );
+    };
+
+    const onChangeCheck = ch => {
+      setAddConstraintsState(prev =>
+        prev.map((c, idx) => {
+          if (i === idx) return { ...c, check: ch };
+          return c;
+        })
+      );
+    };
+
+    const isLast = addConstraintsState.length - 1 === i;
+
+    const existingConstraintName = isLast
+      ? 'new-constraint'
+      : constraints[i] && constraints[i].name;
+
+    const collapsedLabel = () => {
+      if (isLast) {
+        return addConstraintsState.length === 1 ? (
+          <div>
+            <i>(You can add check constraints later as well)</i>
+          </div>
+        ) : null;
+      }
+
+      return (
+        <div>
+          <b>{existingConstraintName}</b>
+        </div>
+      );
+    };
+
+    const expandedLabel = () => {
+      if (isLast) {
+        return null;
+      }
+
+      return (
+        <div>
+          <b>{existingConstraintName}</b>
+        </div>
+      );
+    };
+
+    const expandButtonText = isLast ? 'Add a check constraint' : 'Edit';
+
+    const expandedContent = () => {
+      return (
```

Comment:
We should pull this out into a reusable component. Because the same UI is replicated on the `Modify` page as well. What do you think?